src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp

```diff
@@ -828,6 +792,67 @@ class npc_salanar_the_horseman : public CreatureScript
     }
 };
 
+class spell_deliver_stolen_horse : public SpellScriptLoader
+{
+public:
+    spell_deliver_stolen_horse() : SpellScriptLoader("spell_deliver_stolen_horse") { }
+
+    class spell_deliver_stolen_horse_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_deliver_stolen_horse_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            return ValidateSpellInfo({ SPELL_DELIVER_STOLEN_HORSE, SPELL_EFFECT_STOLEN_HORSE });
+        }
+
+        bool Load() override
+        {
+            _salanarNear = false;
+            return GetCaster()->GetEntry() == NPC_HAVENSHIRE_STALLION || GetCaster()->GetEntry() == NPC_HAVENSHIRE_MARE || GetCaster()->GetEntry() == NPC_HAVENSHIRE_COLT;
+        }
+
+        void HitSalanarCheck(SpellEffIndex /*effIndex*/)
+        {
+            // Can only hit Salanar due to TARGET_UNIT_NEARBY_ENTRY
```

Comment:
the right way would be to abort spells with nearby targets which don't find any target

Reply:
It makes sense but I had no idea on how to do it. 
Which hook is the best for it?

Reply:
@joschiwald could you please help?

Reply:
https://github.com/TrinityCore/TrinityCore/pull/20954

Reply:
All right, I've thought that it was already implemented!
Anyway the second part of the spellscript is needed to implement the cosmetic part of the quest.